src/util/config.rs

```diff
@@ -739,6 +739,132 @@ pub fn check_kernel() -> Vec<ConfigError> {
     Vec::new()
 }
 
+#[derive(Debug, Default)]
+struct FsInfo {
+    tp: String,
+    opts: String,
+    mnt_dir: String,
+    fsname: String,
+}
+
+#[cfg(target_os = "linux")]
+fn get_fsinfo(path: &str, mnt_file: &str) -> Result<FsInfo, ConfigError> {
+    use std::ffi::{CStr, CString};
+    use libc;
+    unsafe {
+        let profile = CString::new(mnt_file).unwrap();
+        let retype = CString::new("r").unwrap();
+        let afile = libc::setmntent(profile.as_ptr(), retype.as_ptr());
+        let mut fs = FsInfo::default();
+        loop {
+            let ent = libc::getmntent(afile);
+            if ent.is_null() {
+                break;
+            }
+            let cur_dir = CStr::from_ptr((*ent).mnt_dir).to_str().unwrap();
+            if path.starts_with(&cur_dir) && cur_dir.len() >= fs.mnt_dir.len() {
+                fs.tp = CStr::from_ptr((*ent).mnt_type).to_str().unwrap().to_owned();
+                fs.opts = CStr::from_ptr((*ent).mnt_opts).to_str().unwrap().to_owned();
+                fs.fsname = CStr::from_ptr((*ent).mnt_fsname)
+                    .to_str()
+                    .unwrap()
+                    .to_owned();
+                fs.mnt_dir = cur_dir.to_owned();
+            }
+        }
+
+        libc::endmntent(afile);
+        if fs.mnt_dir.is_empty() {
+            return Err(ConfigError::Limit(format!(
+                "path:{:?} not find in mountable",
+                path
+            )));
+        }
+        Ok(fs)
+    }
+}
+
+pub fn get_rotational_info(fsname: &str) -> Result<String, ConfigError> {
+    if !fsname.starts_with("/dev/") {
+        return Err(ConfigError::Limit(format!(
+            "fsname:{:?} is not a normal device",
+            fsname
+        )));
+    }
+    let mut dev = fsname.trim_left_matches("/dev/");
+    use std::path::Path;
+    loop {
+        if dev.is_empty() {
+            return Err(ConfigError::Limit(format!(
+                "fs:{} no device find in block",
+                fsname
+            )));
+        }
+        let cur_path = format!("/sys/block/{}", dev);
+        if Path::new(&cur_path).exists() {
+            break;
+        }
+        dev = &dev[0..dev.len() - 1];
```

Comment:
Why?

Reply:
Since `/dev/sda2` and `/dev/sda1` 's block info  maybe in directory `/sys/block/sda/`.